universalModule/software/RoFIDriver/src/rofi_hal.hpp

```diff
@@ -0,0 +1,344 @@
+#pragma once
+#include <functional>
+#include <memory>
+#include "networking.hpp"
+
+namespace rofi::hal {
+
+/**
+ * \brief Proxy for controlling a single joint of RoFI
+ *
+ * Joint cannot be instantiated on its own; you can obtain its instance from
+ * RoFI::getJoint().
+ *
+ * Since the class is only a proxy it can be safely copied and passed around.
+ */
+class Joint {
+public:
+    /**
+     * \brief
+     */
+    enum class Error {
+        Communication, ///< Cannot communicate with the servo motor
+        Hardware ///< Servo motor responded with an error
+    };
+
+    /**
+     *  \brief Interface of the actual implementation - either physical local,
+     *  physical remote or a simulation one.
+     *
+     *  To provide an implementation, simply inherit from this class.
+     *
+     *  For method description, see description of Joint.
+     */
+    class Implementation {
```

Comment:
Add virtual destructor - or (and possibly more preferable) make the destructor protected